### PR TITLE
User settings db

### LINKED ISSUE #14: Dynamic menu notification system
Create a settings system to all users that allow to choose which days and which meals get from the bot.

Example of interfaces (in buttons):

| Giorni    | Pranzo | Cena |
| --------- | ------ | ---- |
| Lunedì    | ✅     | ✅   |
| Martedì   | ✅     | ✅   |
| Mercoledì | ✅     | ✅   |
| Giovedì   | ✅     | ✅   |
| Venerdì   | ✅     | ✅   |
| Sabato    | ❌     | ❌   |
| Domenica  | ❌     | ❌   |

By clicking on ❌ it will switch to ✅ and vice versa.

In this case, the user will receive a notification for lunch and dinner (15 or 30 minutes before the open time) on all days except for Saturday and Sunday.

By default, this will be only lunch from Mon-Friday.

This task consists of:
- creating a database to save all the information per user
- creating a periodic job that checks the database and sends the notification to all users registered

This is a base, in the future we could add more features to this system.

### PR DESCRIPTION
**Pull request by infra-blue**
I added an sqlite database for storing user preferences about mensa' s menù notification.
I added a class to easily access and manage the database.
closes #14

### CODE REVIEW

File: module/data/menu_keys.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+saturday_lunch_false = InlineKeyboardButton("❌", callback_data="21")
+saturday_lunch_true = InlineKeyboardButton("✅", callback_data="22")
+saturday_dinner_false = InlineKeyboardButton("❌", callback_data="23")
+saturday_dinner_true = InlineKeyboardButton("✅", callback_data="24")
+
+sunday_lunch_false = InlineKeyboardButton("❌", callback_data="25")
+sunday_lunch_true = InlineKeyboardButton("✅", callback_data="26")
+sunday_dinner_false = InlineKeyboardButton("❌", callback_data="27")
+sunday_dinner_true = InlineKeyboardButton("✅", callback_data="28")
```
Comment (Helias) [inline module/data/menu_keys.py:54]: all this code is really redundant we have to follow the DRY principle and wrap it into an array if we really need it
but I don't think that we need this

File: module/data/user_settings.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                AND thursday_dinner = 0
+
+                AND friday_lunch = 0
+                AND friday_dinner = 0
+
+                AND saturday_lunch = 0
+                AND saturday_dinner = 0
+
+                AND sunday_lunch = 0
+                AND sunday_dinner = 0
```
Comment (Helias) [inline module/data/user_settings.py:128]: not sure if we need this list of `AND`s

File: module/data/user_settings.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            cur = con.cursor()
+            cur.execute("""SELECT chat_id
+            FROM user_settings
+            WHERE chat_id = (?)""", (chat_id, ))
+
+            user_exists = cur.fetchall()
+            if len(user_exists) == 0:
+                cur.execute("INSERT INTO user_settings (chat_id) values (?)", (chat_id, ))
+
+    def setMeal(self, chat_id: int, day: DAYS, meal: MEALS) -> None:
```
Comment (Helias) [inline module/data/user_settings.py:73]: `set_meal`

File: module/data/user_settings.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                friday_dinner INTEGER(1) NOT NULL DEFAULT 0,
+
+                saturday_lunch INTEGER(1) NOT NULL DEFAULT 0,
+                saturday_dinner INTEGER(1) NOT NULL DEFAULT 0,
+
+                sunday_lunch INTEGER(1) NOT NULL DEFAULT 0,
+                sunday_dinner INTEGER(1) NOT NULL DEFAULT 0
+                )""")
+
+    def getUsers(self) -> list:
```
Comment (Helias) [inline module/data/user_settings.py:36]: `get_users`

File: module/data/user_settings.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
+import sqlite3
+from .constants import DB_PATH
+from .constants import DAYS, MEALS, VALID_DAYS, VALID_MEALS
+
+
+class UserSettings:
```
Comment (Helias) [inline module/data/user_settings.py:6]: :+1:

File: module/commands/menu.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    return
+
+def menu_settings(update: Update, context: CallbackContext) -> None:
+    menu_set = UserSettings()
+
+#    keyboard = [
+#        GIORNI_PRANZO_CENA,
+#        [
+#            InlineKeyboardButton("Lunedì", callback_data="0"),
+#            InlineKeyboardButton("❌", callback_data="1"),
```
Comment (Helias) [inline module/commands/menu.py:18]: we could put `"❌"` in a constant variable like `CROSS`

File: module/data/user_settings.py (hunk 2 above)
Comment (InfraBlue0) [reply module/data/user_settings.py:128]: I'm not sure too, I'll search a better way to do the same thing

File: module/commands/menu.py (hunk 6 above)
Comment (InfraBlue0) [reply module/commands/menu.py:18]: ok, perfect

File: module/data/user_settings.py (hunk 2 above)
Comment (infra-blue) [reply module/data/user_settings.py]: By the way, I removed it, the cancelation of a record may cause problems with the dynamic generation of the keyboard

File: module/commands/menu_settings.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+from module.data.user_settings import UserSettings
+from module.data.constants import DB_PATH
+from module.data.constants import CROSS, CHECK
+
+def generate_keyboard(chat_id: int) -> list:
+    menu_sett = UserSettings()
+    settings = (menu_sett.get_user_settings(chat_id))
+    keyboard = [
+        [
+            InlineKeyboardButton("Giorni", callback_data = "0"),
```
Comment (Helias) [inline module/commands/menu_settings.py]:
```python
EMPTY = "0" # oppure potresti mettere anche EMPTY=""
...
            InlineKeyboardButton("Giorni", callback_data=EMPTY),
```

File: module/commands/menu_settings.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+            InlineKeyboardButton("Pranzo", callback_data = "0"),
+            InlineKeyboardButton("Cena", callback_data = "0")
+        ],
+
+        [
+            InlineKeyboardButton("Lunedì", callback_data = "0"),
+            InlineKeyboardButton(CROSS if settings[1] == 0 else CHECK, callback_data = "monday_lunch"),
```
[the remark this answers is not in the document]
Comment (infra-blue) [reply module/commands/menu_settings.py]: Geniale! Grazie

File: module/data/user_settings.py [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+            cur = con.cursor()
+            result = cur.execute("""SELECT *
+            FROM user_settings
+            WHERE chat_id = (?)
+            """, (chat_id, )
+            )
+            temp = (result.fetchall())[0]
+            settings = [None] * 14
+            for var in range(0, 14):
+                settings[var] = temp[var + 1]
```
Comment (Helias) [inline module/data/user_settings.py:93]:
```python
settings = [ tmp[var+1] for var in range(14) ]
```

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
I left a minor comment, for the rest it looks good to me.
Keep in mind that you have to fix the pipeline reports, so the pylint warnings.

**Pull request merged**